src/ophyd_async/core/_hdf_dataset.py

```diff
@@ -67,7 +66,7 @@ def __init__(
                 parameters={
                     "dataset": ds.dataset,
                     "swmr": ds.swmr,
-                    "multiplier": ds.multiplier,
+                    "shape": ds.shape,
```

Comment:
I forgot what we decided here - did we want the shape in the resource doc, or just in the data key that you can get a reference to from the resource doc? Chunking info definitely makes sense in the resource doc since certain data formats won't really have chunking as an option.

Reply:
I think we decided just in the descriptor